wireapp/wire-android
# fix: 1:1 calls can not be established on staging environment (WPB-9359) - approach 3

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-9359" title="WPB-9359" target="_blank"><img alt="Bug" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10803?size=medium" />WPB-9359</a>  [Android] 1:1 calls can not be established on staging environment
  </td></table>
  <br />
 








…ta build

----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

1:1 calls can not be established.

### Causes (Optional)

After extracting calls to another Activity, the app cannot observe its lifecycle state properly.

### Solutions

Move observing from `WireActivity` to `WireApplication` so that it's in one place and the state is consistent across the app.

### Testing

#### How to Test

STR:
- Login to staging env on android as user A
- Login to saging env on web as user B
- Start a 1:1 call on android as user A
- Pick it up on web as user B

Expected: 
- 1:1 call should be established

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
